examples/siamese_network/siamese_network.py

```diff
@@ -0,0 +1,284 @@
+        anchor_out = model(anchor_image)
+        positive_out = model(positive_image)
+        negative_out = model(negative_image)
+        test_loss = criterion(anchor_out, positive_out, negative_out)
+        return test_loss
+
+    # create evaluator engine and attach test step
+    evaluator = Engine(test_step)
+
+    # attach progress bar to evaluator
+    pbar = ProgressBar()
+    pbar.attach(evaluator)
+
+    # attach various handlers to evaluator engine
+    @evaluator.on(Events.ITERATION_COMPLETED(every=log_interval))
+    def log_testing_results(engine):
+        nonlocal average_test_loss
+        average_test_loss += engine.state.output
+        print(f"Test Epoch: {engine.state.epoch} Test Loss: {engine.state.output: .5f}")
+
+    evaluator.add_event_handler(Events.ITERATION_STARTED, lr_scheduler)
+
+    # run evaluator engine
+    evaluator.run(test_loader)
+
+    # print average loss over test dataset
+    print(f"Average Test Loss: {average_test_loss/len(test_loader.dataset): .7f}")
+
+
+def main():
+    # adds training defaults and support for terminal arguments
+    parser = argparse.ArgumentParser(description="PyTorch Siamese network Example")
+    parser.add_argument(
+        "--batch-size", type=int, default=64, metavar="N", help="input batch size for training (default: 64)"
+    )
+    parser.add_argument(
+        "--test-batch-size", type=int, default=200, metavar="N", help="input batch size for testing (default: 1000)"
+    )
+    parser.add_argument("--epochs", type=int, default=10, metavar="N", help="number of epochs to train (default: 14)")
+    parser.add_argument("--lr", type=float, default=1.0, metavar="LR", help="learning rate (default: 1.0)")
+    parser.add_argument(
+        "--gamma", type=float, default=0.95, metavar="M", help="Learning rate step gamma (default: 0.7)"
+    )
+    parser.add_argument("--no-cuda", action="store_true", default=False, help="disables CUDA training")
+    parser.add_argument("--no-mps", action="store_true", default=False, help="disables macOS GPU training")
+    parser.add_argument("--dry-run", action="store_true", default=False, help="quickly check a single pass")
+    parser.add_argument("--seed", type=int, default=1, metavar="S", help="random seed (default: 1)")
+    parser.add_argument(
+        "--log-interval",
+        type=int,
+        default=10,
+        metavar="N",
+        help="how many batches to wait before logging training status",
+    )
+    parser.add_argument("--save-model", action="store_true", default=False, help="For Saving the current Model")
+    args = parser.parse_args()
+
+    # set device
+    device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+
+    # data loading
+    train_dataset = MatcherDataset("../data", train=True, download=True)
+    test_dataset = MatcherDataset("../data", train=False)
+    train_loader = DataLoader(train_dataset, shuffle=True, batch_size=args.batch_size)
```

Comment:
Please add `num_workers` as argparse argument and set default value to 4 (num_workers=4)